### PR TITLE
Validation error not displayed for Incorrect eth address in "Sent To" page #19599

### LINKED ISSUE #19599: Validation error not displayed for Incorrect eth address in "Sent To" page
### Steps:
1. Open wallet
2. Go to 'send to' page 
3. Paste valid eth account address
4. Update  account address making it as invalid

### Actual result:
The "Confirm" button is shown as enabled, and an error is shown after tapping on it when an invalid ETH address is in the "sent to" input field.

https://github.com/status-im/status-mobile/assets/52490791/f6bb8611-217f-448d-97f2-a2fcd951cda3

### Expected result:
- Only in case of correct multichain or regular eth address entering/pasting the 'confirm' buttons should be enabled
- The validation "This is not valid Ethereum address" error message should be shown upon entering an invalid ETH address.

discussed with design team https://discord.com/channels/624634427930312714/1221764463732981790/1225799935065198693

### OS: 
IOS, Android
### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17
### Logs

### PR DESCRIPTION
**Pull request by mmilad75**
fixes #19599

https://github.com/status-im/status-mobile/assets/55688834/c22c4eb2-7a5d-4285-91cf-75331a678267

### CODE REVIEW

File: src/status_im/contexts/wallet/events.cljs [hunk 1]
```diff
@@ -326,55 +326,26 @@
...
- (fn [{:keys [db]} [_]]
+ (fn [{:keys [db]}]
    {:db (assoc-in db [:wallet :ui :search-address :valid-ens-or-address?] true)}))
 
+(rf/reg-event-fx :wallet/address-validation-failed
+ (fn [{:keys [db]}]
+   {:db (assoc-in db [:wallet :ui :search-address :valid-ens-or-address?] false)}))
+
 (rf/reg-event-fx :wallet/validate-address
- (fn [{:keys [db]} [address]]
```
Comment (mmilad75) [inline src/status_im/contexts/wallet/events.cljs:349]: These events were implemented for mock purposes. Now, they're not needed anymore. So I deleted them

File: src/status_im/contexts/wallet/events.cljs [hunk 2]
```diff
@@ -326,55 +326,26 @@
...
 (rf/reg-event-fx :wallet/address-validation-success
- (fn [{:keys [db]} [_]]
+ (fn [{:keys [db]}]
    {:db (assoc-in db [:wallet :ui :search-address :valid-ens-or-address?] true)}))
 
+(rf/reg-event-fx :wallet/address-validation-failed
+ (fn [{:keys [db]}]
+   {:db (assoc-in db [:wallet :ui :search-address :valid-ens-or-address?] false)}))
+
 (rf/reg-event-fx :wallet/validate-address
```
Comment (vkjr) [inline src/status_im/contexts/wallet/events.cljs]: Here the event is used as a container for validation logic which is not ideal. I'd suggest moving logic to a function instead and use it in place of event, wdyt?

Comment (mmilad75) [reply src/status_im/contexts/wallet/events.cljs]: Done

File: src/status_im/contexts/wallet/send/select_address/view.cljs [hunk 3]
```diff
@@ -32,7 +33,15 @@
...
-          contacts                 (rf/sub [:contacts/active])]
+          contacts                 (rf/sub [:contacts/active])
+          validate-address         (fn [address]
+                                     (debounce/debounce-and-dispatch
+                                      (if (and (> (count address) 0) 
+                                               (not (or (validation/ens-name? address) 
+                                                        (validation/eth-address? address))))
+                                        [:wallet/address-validation-failed address]
+                                        [:wallet/address-validation-success address])
+                                      300))]
```
Comment (OmarBasem) [inline src/status_im/contexts/wallet/send/select_address/view.cljs]: Maybe we can remove this logic from the view, and just `debounce-and-dispatch` an event with the necessary args

Comment (mmilad75) [reply src/status_im/contexts/wallet/send/select_address/view.cljs]: It was like that before, but @vkjr suggested this

Comment (OmarBasem) [reply src/status_im/contexts/wallet/send/select_address/view.cljs]: Oh I see.. considering the validation logic then maybe better to leave it in the view 👍

**Review by OmarBasem — APPROVED** [review]

**Review by vkjr — APPROVED** [review]

File: src/status_im/contexts/wallet/send/select_address/view.cljs [hunk 4]
```diff
@@ -32,7 +33,15 @@
           recipient                (rf/sub [:wallet/wallet-send-recipient])
           recipient-plain-address? (= send-address recipient)
           valid-ens-or-address?    (rf/sub [:wallet/valid-ens-or-address?])
-          contacts                 (rf/sub [:contacts/active])]
+          contacts                 (rf/sub [:contacts/active])
+          validate-address         (fn [address]
```
Comment (vkjr) [inline src/status_im/contexts/wallet/send/select_address/view.cljs]: Maybe it can be taken out of component for better readability, wdyt?

Comment (mmilad75) [reply src/status_im/contexts/wallet/send/select_address/view.cljs]: Done

**Comment by VolodLytvynenko** [comment]
hi @mmilad75 Thank you for PR. No issues from my side. Ready to be merged

**Pull request merged**